cpp/iedriver/IECommandExecutor.cpp

```diff
@@ -438,6 +439,8 @@ unsigned int WINAPI IECommandExecutor::ThreadProc(LPVOID lpParameter) {
         LOG(DEBUG) << "Returned from DestroyWindow()";
         break;
       } else {
+        static std::mutex messageLock;
+        std::lock_guard<std::mutex> lock(messageLock);
```

Comment:
I'm not a Win32 hacker, so I'm not sure how this fixes the stability issue. The docs for `GetMessage`, `TranslateMessage`, and `DispatchMessage` suggest they only operate on the current thread. Is this something to do with multiple threads attempting to post messages to the same `HWND`?

Reply:
It appears to me that this code is indeed called from multiple threads.  In a typical win32 program these go on what I call the "UI thread" but that's not the context they're being used in here.  Perhaps I'm wrong and this is only called in one thread, in which case this PR is useless, but I believe there are multiple threads using this.

Reply:
Huh, so although the docs suggest that these methods are thread safe, they're not really because the normal use case is that they're only used on the UI thread? Interesting.

Your observed increase in stability is a Good Thing, and I'm happy to trust your reasoning :)